### ISSUE TITLE
LogInstrumentation logs long stack trace on normal behavior

### ISSUE THREAD

**Issue by TWiStErRob** [newrelic/newrelic-android-agent#280]
## Description

There's log spam when code calls `Log.x("TAG", "message", null")`.

## Steps to Reproduce

```java
android.util.Log.i("TEST", "null exception 1");
android.util.Log.i("TEST", "null exception 2", null);
```

## Expected Behavior

No stack trace printed.
```
2024-09-25 16:18:39.827 21454-21454 TEST                    com.example       I  null exception 1
2024-09-25 16:18:39.827 21454-21454 TEST                    com.example       I  null exception 2
```

## Actual Behavior

```
2024-09-25 15:26:29.864 24747-24747 TEST                    com.example       I  null exception 1
2024-09-25 15:26:29.864 24747-24747 newrelic                com.example       I  newrelic: {level=INFO,tag=TEST,message=null exception 1}
2024-09-25 15:26:29.864 24747-24747 TEST                    com.example       I  null exception 2
2024-09-25 15:26:29.865 24747-24747 newrelic                com.example       I  newrelic: java.lang.IllegalArgumentException
    at com.newrelic.agent.android.logging.MessageValidator.validate(MessageValidator.java:72)
    at com.newrelic.agent.android.logging.LogReporting$AgentLogger.logAll(LogReporting.java:190)
    at com.newrelic.agent.android.instrumentation.LogInstrumentation.i(LogInstrumentation.java:78)
    at com.example.MyActivity.onCreate(MyActivity.kt:50)
    at android.app.Activity.performCreate(Activity.java:8305)
    at android.app.Activity.performCreate(Activity.java:8284)
    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417)
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3626)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3782)
    at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
    at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:201)
    at android.os.Looper.loop(Looper.java:288)
    at android.app.ActivityThread.main(ActivityThread.java:7872)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
 {level=INFO,tag=TEST,message=null exception 2}
```

## Your Environment

* NewRelic com.newrelic.agent.android:android-agent
  * 🔴 7.6.0 stacktrace
  * 🔴 7.5.1 stacktrace
  * 🔴 7.4.0 stacktrace
  * 🟢 7.3.0 no stacktrace
  * 🟢 7.2.0 no stacktrace
* API 33 (probably not relevant)

## Additional context

Why is this a problem? There are many logging abstractions out there, and it's easy to imagine them being written in Kotlin. This means that we can have signatures like
```kotlin
fun Logger.info(message: String, throwable: Throwable? = null) {
    Log.i(this.tag, message, throwable)
}
```
notice the default argument. That means the 2nd variant in repro is always going to be called when the user does `this.logger.info("Hello")`, meaning for every non-erroring log line NewRelic will print a full stack trace, because it's "missing" an exception.

Even the signature of `Log.i` allows for a `@Nullable Throwable`, so `LogInstrumentation` should handle that:

https://github.com/newrelic/newrelic-android-agent/blob/2265f8f57a1f8f96f170fe9b69d5fef7f156bb3a/agent-core/src/main/java/com/newrelic/agent/android/logging/LogReporting.java#L190-L194

https://github.com/newrelic/newrelic-android-agent/blob/2265f8f57a1f8f96f170fe9b69d5fef7f156bb3a/agent-core/src/main/java/com/newrelic/agent/android/logging/MessageValidator.java#L71-L73

Note: in the repro I removed all the unnecessary baggage (3rd party lib, Kotlin), just focused on the bytecode that minimally triggers this behavior.

A real case of this "imaginary logger" can be found at [Interface definition with default args](https://github.com/ChuckerTeam/chucker/blob/4.0.0/library/src/main/kotlin/com/chuckerteam/chucker/internal/support/Logger.kt) and [implementation with call to Log.i](https://github.com/ChuckerTeam/chucker/blob/4.0.0/library/src/main/kotlin/com/chuckerteam/chucker/api/Chucker.kt#L79).

<details><summary>Stack trace from Chucker's usage</summary>

```
java.lang.IllegalArgumentException
    at com.newrelic.agent.android.logging.MessageValidator.validate(MessageValidator.java:72)
    at com.newrelic.agent.android.logging.LogReporting$AgentLogger.logAll(LogReporting.java:190)
    at com.newrelic.agent.android.instrumentation.LogInstrumentation.i(LogInstrumentation.java:78)
    at com.chuckerteam.chucker.api.Chucker$logger$1.info(Chucker.kt:82)
    at com.chuckerteam.chucker.internal.support.Logger$Companion.info(Logger.kt:14)
    at com.chuckerteam.chucker.internal.support.Logger$DefaultImpls.info$default(Logger.kt:6)
    at com.chuckerteam.chucker.internal.support.RequestProcessor$decodePayload$1.invoke(RequestProcessor.kt:75)
    at com.chuckerteam.chucker.internal.support.RequestProcessor$decodePayload$1.invoke(RequestProcessor.kt:73)
    at kotlin.sequences.TransformingSequence$iterator$1.next(Sequences.kt:210)
    at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:170)
    at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:194)
    at kotlin.sequences.SequencesKt___SequencesKt.firstOrNull(_Sequences.kt:168)
    at com.chuckerteam.chucker.internal.support.RequestProcessor.decodePayload(RequestProcessor.kt:81)
    at com.chuckerteam.chucker.internal.support.RequestProcessor.processPayload(RequestProcessor.kt:64)
    at com.chuckerteam.chucker.internal.support.RequestProcessor.process(RequestProcessor.kt:22)
    at com.chuckerteam.chucker.api.ChuckerInterceptor.intercept(ChuckerInterceptor.kt:76)
    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
    at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.kt:221)
    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
    at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
    at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
    at java.lang.Thread.run(Thread.java:1012)
```

</details>

Additionally we could argue that in general double-printing logs (even when they're not printing stack traces) is noisy in LogCat.

**Comment by ndesai-newrelic** [comment]
@TWiStErRob thanks for your feedback, we will look into it.

**Comment by TWiStErRob** [comment]
@ndesai-newrelic I see a merged PR, does this mean it's fixed and we're waiting for a release?

**Comment by ndesai-newrelic** [comment]
the issue is fixed in latest android agent release 7.6.2.
